File: Data Structures and Algorithm Analysis in C/Unit7/Selection_Sort.c
/*
*选择排序(升序)
*/

void Selection_Sort(const int *arr, int N)
{
	int i, j, k, tmp;

	for(i = 0; i < N - 1; i++){
		k = i;
		for(j = i + 1; j < N; j++){
			if(arr[j] < arr[i]){
				k = j;
			}
		}

		if(k != i){
			tmp = arr[i];
			arr[i] = arr[k];
			arr[k] = tmp;
		}
	}
}

File: Algorithms And Design in Java/algorithm/sort/SelectionSort.java
package com.deceiver.algorithm.sort;

import java.util.Arrays;

public class SelectionSort {

    private static void sort(int[] arr){
        int len = arr.length;
        for(int i = 0; i < len-1; i++){
            int k = i;
            for(int j = i + 1; j < len; j++){
                if(arr[j] < arr[k]){
                    k = j;
                }
            }
            if(k != i){
                arr[k] ^= arr[i];
                arr[i] ^= arr[k];
                arr[k] ^= arr[i];
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = AlgorithmUtils.getRandomArray();
        sort(arr);
        AlgorithmUtils.printArray(arr);
    }
}
